pkg/gadgets/snapshot/process/tracer/tracer_test.go

```diff
@@ -0,0 +1,178 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+	"testing"
+
+	containerutils "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/internal/test"
+	snapshotProcessTypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/snapshot/process/types"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type collectorFunc func(config *Config, enricher gadgets.DataEnricher) ([]*snapshotProcessTypes.Event, error)
+
+func BenchmarkEBPFTracer(b *testing.B) {
+	benchmarkTracer(b, runeBPFCollector)
+}
+
+func BenchmarkProcfsTracer(b *testing.B) {
+	benchmarkTracer(b, runProcfsCollector)
+}
+
+func benchmarkTracer(b *testing.B, runCollector collectorFunc) {
+	utilstest.RequireRoot(b)
+
+	for n := 0; n < b.N; n++ {
+		_, err := runCollector(&Config{}, nil)
+		if err != nil {
+			b.Fatalf("benchmarking collector: %s", err)
+		}
+	}
+}
+
+func TestEBPFTracer(t *testing.T) {
+	testTracer(t, runeBPFCollector)
+}
+
+func TestProcfsTracer(t *testing.T) {
+	testTracer(t, runProcfsCollector)
+}
+
+func testTracer(t *testing.T, runCollector collectorFunc) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	type testDefinition struct {
+		getTracerConfig func(info *utilstest.RunnerInfo) *Config
+		runnerConfig    *utilstest.RunnerConfig
+		generateEvent   func() (int, error)
+		validateEvent   func(t *testing.T, info *utilstest.RunnerInfo, sleepPid int, events []snapshotProcessTypes.Event)
+	}
+
+	for name, test := range map[string]testDefinition{
+		"captures_all_events_with_no_filters_configured": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *Config {
+				return &Config{}
+			},
+			generateEvent: generateEvent,
+			validateEvent: utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, sleepPid int) *snapshotProcessTypes.Event {
+				return &snapshotProcessTypes.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Command:   "sleep",
+					Pid:       sleepPid,
+					Tid:       sleepPid,
+					MountNsID: info.MountNsID,
+				}
+			}),
+		},
+		"captures_no_events_with_no_matching_filter": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *Config {
+				// We can't use 0 as the mntnsid becuase the tracer will collect
+				// some process that have this as mntnsid. Then, we set the network
+				// namespace inode id to be sure there is not any mount namespace
+				// using that same inode.
```

Comment:
What about using the mntns of `os.Gettid()` (the current go routine is not the thread of the runner)?

Reply:
I don't think it'll work because the goal is not to capture processes at all. If we use the inode of a valid namespace will capture the processes running there.